File: app/admin/video.rb
ActiveAdmin.register Video do
  form do |_f|
    input :user
    input :tournament
    input :aff_team
    input :neg_team
    input :debate_type, as: :select, collection: Video.debate_types.keys.to_a
    input :debate_level, as: :select, collection: Video.debate_levels.keys.to_a
    input :thumbnail
    input :live_now
    input :is_featured
    actions
  end

  controller do
    def permitted_params
      params.permit! # allow all parameters
    end
  end
end

File: app/assets/components/Videos/store/records/video.js
import { List, Record } from 'immutable';
import Team from './team';
import Tournament from './tournament';

const defaultVideo = {
  id: null,
  debateType: null,
  debateLevel: null,
  thumbnail: null,
  liveNow: false,
  isFeatured: false,
  tournamentId: null,
  tournament: new Tournament(),
  affTeamId: null,
  affTeam: new Team(),
  negTeamId: null,
  negTeam: new Team(),
  tags: List(),
  tagsVideos: null,
};

export default class Video extends Record(defaultVideo) {
  getTitle() {
    return `${this.tournament.getYearAndName()}: ${this.affTeam.getTeamCode()} vs. ${this.negTeam.getTeamCode()}`;
  }
}
